queries/buildings.jinja2

```diff
@@ -1,13 +1,20 @@
 SELECT
-    name,
+    -- drop name when the item is a POI - in that case the name
+    -- probably refers to the POI. instead, we can use the house
+    -- name tag, if it has been set. likewise for landuse areas
+    -- where we might want to label the landuse rather than the
+    -- building.
+    CASE WHEN mz_poi_min_zoom IS NULL AND
+              (mz_is_landuse IS NULL OR mz_is_landuse = FALSE)
+         THEN name
+         ELSE "addr:housename"
+    END AS name,
     way_area::bigint AS area,
     osm_id AS __id__,
-    "building:part",
+    'openstreetmap.org' AS source,
```

Comment:
In the not too distant future we may also offer Polygon City building data.
So we can either start doing this now, or then.

On Thu, Oct 15, 2015 at 9:17 AM, Matt Amos <notifications@github.com> wrote:

> In queries/buildings.jinja2
> <https://github.com/mapzen/vector-datasource/pull/307#discussion_r42145658>
> :
>
> >      way_area::bigint AS area,
> >      osm_id AS __id__,
> > -    "building:part",
> > +    'openstreetmap.org' AS source,
>
> I don't know. I *thought* I was adding it for compatibility with the
> other layers, but it seems that we don't include source for the
> boundaries, earth, POIs or transit layers either. Transit and POIs are
> currently OSM-only, but earth and boundaries do mix data sources...
>
> @nvkelso <https://github.com/nvkelso> Should we be annotating the source
> for *all* features, or just features in layers which aren't single-source?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mapzen/vector-datasource/pull/307/files#r42145658>.
>